clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -422,21 +423,17 @@ public synchronized Set<TopicPartition> updatePartitionLeadership(Map<TopicParti
             updatePartitionMetadata.add(updatedMetadata);
 
             lastSeenLeaderEpochs.put(partition, newLeader.epoch.get());
+
+            final String updatedTopic = updatedMetadata.topic();
+            if (this.metadataSnapshot.topicIds().containsKey(updatedTopic))
+                topicIdsForUpdatedTopics.put(updatedTopic, this.metadataSnapshot.topicIds().get(updatedTopic));
```

Comment:
This is bit uglier, but it avoids the second hash map lookup.

```suggestion
            Uuid topicUuid = this.metadataSnapshot.topicIds().getOrDefault(updatedTopic, Uuid.ZERO_UUID);

            if (!topicUuid.equals(Uuid.ZERO_UUID))
                topicIdsForUpdatedTopics.put(updatedTopic, topicUuid);
```

Reply:
Makes sense to me. I think a similar approach can be taken in MetadataSnapshot in the lines below

`if (this.topicIds.containsKey(entry.getKey().topic()))`
`    newTopicIds.putIfAbsent(entry.getKey().topic(), this.topicIds.get(entry.getKey().topic()));`